### PR TITLE
CPU consumption fix, missed Test Case ID bypass fix

### PR DESCRIPTION
**Pull request by HardNorth**
- CPU consumption fix
- Test Case ID bypass fix
- Ability to bypass `None` status, since it's legal in the latest versions of Report Portal
- `ExternalIssue` class `btsUrl` field name fix
- Add priority for `ControlCommand` class

### CODE REVIEW

File: reportportal_client/core/worker.py [hunk 1]
```diff
@@ -43,22 +43,16 @@ def is_stop_cmd(self):
 class APIWorker(object):
     """Worker that makes non-blocking HTTP requests to the Report Portal."""
```
Comment (iivanou) [inline reportportal_client/core/worker.py:44]: It's not non-blocking anymore.

File: reportportal_client/core/worker.py [hunk 2]
```diff
@@ -43,22 +43,16 @@ def is_stop_cmd(self):
...
-        """Initialize instance attributes.
-
-        :param cmd_queue:  Queue for the control commands
-        :param data_queue: Queue for the RP requests to process
-        """
-        self._cmd_queue = cmd_queue
-        self._data_queue = data_queue
+    def __init__(self):
+        """Initialize instance attributes."""
+        self._queue = queue.Queue()
```
Comment (iivanou) [inline reportportal_client/core/worker.py:48]: We need the PriorityQueue here. Our requests have their own priorities. Revert the changes and modify the code to use only one queue instead of two. The queue should be passed into the constructor.

File: reportportal_client/core/worker.py [hunk 3]
```diff
@@ -43,22 +43,16 @@ def is_stop_cmd(self):
...
         self._thread = None
         self.name = self.__class__.__name__
 
     def _command_get(self):
-        """Get control command from the control queue."""
+        """Get command from the queue."""
         try:
-            cmd = self._cmd_queue.get_nowait()
-            logger.debug('[%s] Received {%s} command', self.name, cmd)
+            cmd = self._queue.get(timeout=1)
```
Comment (iivanou) [inline reportportal_client/core/worker.py:55]: Reduce timeout to 0.1

**Review by iivanou — CHANGES_REQUESTED** [review]

File: reportportal_client/core/worker.py (hunk 2 above)
Comment (HardNorth) [reply reportportal_client/core/worker.py:48]: Maybe I messed that, but I don't see where we use any feature of PriorityQueue?

**Comment by HardNorth** [comment]
+ 'evaluate_status' function in helpers

**Review by iivanou — CHANGES_REQUESTED** [review]
You do not need to add all of this(the commit: Add processing priority), if you leave the previous code where we used the PriorityQueue. The protocol is already implemented [here](https://github.com/reportportal/client-Python/blob/master/reportportal_client/core/rp_requests.py#L87).

File: reportportal_client/core/worker.py [hunk 4]
```diff
@@ -39,51 +43,52 @@ def is_stop_cmd(self):
         """Verify if the command is the stop one."""
         return self in (ControlCommand.STOP, ControlCommand.STOP_IMMEDIATE)
 
+    @property
+    def priority(self):
+        """Get the priority of the command."""
+        if self is ControlCommand.STOP_IMMEDIATE:
+            return Priority.PRIORITY_IMMEDIATE
+        else:
```
Comment (iivanou) [inline reportportal_client/core/worker.py:51]: This line can be removed.

**Comment by iivanou** [comment]
The rest of the code looks good to me.

File: reportportal_client/core/worker.py [hunk 5]
```diff
@@ -39,51 +43,49 @@ def is_stop_cmd(self):
         """Verify if the command is the stop one."""
         return self in (ControlCommand.STOP, ControlCommand.STOP_IMMEDIATE)
 
+    @property
+    def priority(self):
+        """Get the priority of the command."""
+        return Priority.PRIORITY_LOW
```
[the remark this answers is not in the document]
Comment (HardNorth) [reply reportportal_client/core/worker.py:49]: Oh, it shows several lines of code to me, so I thought you don't want higher priority for the command. OK, I'll fix it.

**Review by iivanou — APPROVED** [review]
